libcontainer/container_linux.go

```diff
@@ -229,6 +239,30 @@ func (c *Container) Exec() error {
 	return c.exec()
 }
 
+// terminate is to kill the container's init/exec process when got failure.
+func (c *Container) terminate(process *Process) {
+	if process.ops == nil {
+		return
+	}
+	if process.Init {
+		if err := ignoreTerminateErrors(process.ops.terminate()); err != nil {
+			logrus.WithError(err).Warn("unable to terminate initProcess")
+		}
+		if err := c.cgroupManager.Destroy(); err != nil {
+			logrus.WithError(err).Warn("unable to destroy cgroupManager")
+		}
+		if c.intelRdtManager != nil {
+			if err := c.intelRdtManager.Destroy(); err != nil {
+				logrus.WithError(err).Warn("unable to destroy intelRdtManager")
+			}
+		}
```

Comment:
With the info in the PR and this github diff, it's not obvious why this makes sense. Can you please elaborate why before in many places we were not calling all of this and now we do? For example, the cgroup manager destroy wasn't called in all the code-paths before.

Was it missing? It wasn't needed but it's idempotent so it is fine?

Reply:
I think whether these destroy methods are called or not both are OK.
We should not have different results in one function without a specific reason, for example: in `container.Run()`.

Reply:
And I think we should not destroy them in before, because runc doesn’t delete the failure container automatically, users must have to use ‘runc delete’ to destroy the failure container created by ‘runc create’ or ‘runc run’. How about remove these destroy methods? I think it has no compatibility problems. WDYT?

Reply:
Sorry, not sure I followed. Why do you want to remove these methods?

Also, what is the state the container is left? The delete operation must only work for stopped containers: https://github.com/opencontainers/runtime-spec/blob/main/runtime.md#delete

Reply:
> Sorry, not sure I followed. Why do you want to remove these methods?

Yes, these destroy methods can't be removed, because we should destroy the cgroup & intelRdt manager manually if we haven't saved container's state yet.